### PR TITLE
Documentation Cleanup

### PR DESCRIPTION
**Pull request by rifelpet**
Throughout my learning of Kops I've updated the docs on anything that was out of date or missing. Also fixed a few typos.

This should close out #2921

### CODE REVIEW

File: docs/instance_groups.md [hunk 1]
```diff
@@ -122,25 +120,20 @@ So the procedure is:
...
 
 ```
 metadata:
   creationTimestamp: "2016-07-10T15:47:14Z"
   name: nodes
 spec:
-  machineType: m3.medium
-  maxPrice: "0.1"
+  machineType: t2.medium
+  maxPrice: "0.01"
```
Comment (chrislovecnm) [inline docs/instance_groups.md]: you cannot do spot instances on t2 types :(

File: docs/labels.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Both are specified at the InstanceGroup level.
 
 A nice use for cloudLabels is to specify [AWS cost allocation tags](http://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/cost-alloc-tags.html).
 
-A good use for nodeLables is to implement [nodeSelector labels](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#step-two-add-a-nodeselector-field-to-your-pod-configuration).
```
Comment (chrislovecnm) [inline docs/labels.md]: Can we add a link to this doc from the instance group document??

Comment (rifelpet) [reply docs/labels.md]: done 👍

File: docs/instance_groups.md (hunk 1 above)
Comment (chrislovecnm) [reply docs/instance_groups.md]: Cool, that is new.

**Comment by chrislovecnm** [comment]
/lgtm

Again thanks for the awesome updates.  Love people that do documentation.

**Pull request merged**